lib/improver/generate_ancillaries/generate_ancillary.py

```diff
@@ -0,0 +1,249 @@
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string"""
+        result = ('<CorrectLandSeaMask>')
+        return result
+
+    @staticmethod
+    def process(standard_landmask):
+        """Read in the interpolated landmask and assign values round
+            values < 0.5 to False and >=0.5 to True.
+
+        Parameters
+        ----------
+        standard_landmask: cube
+            input landmask on standard grid.
+
+        Returns
+        -------
+        standard_landmask : cube
+            output landmask of boolean values.
+        """
+        mask_sea = np.ma.masked_less(standard_landmask.data, 0.5).mask
+        standard_landmask.data[mask_sea] = False
+        mask_land = np.ma.masked_greater(standard_landmask.data, 0.).mask
+        standard_landmask.data[mask_land] = True
+        return standard_landmask
+
+
+class GenerateOrographyBandAncils(object):
+    """
+    Generate topographic band ancillaries for the standard grids.
+
+    Reads StaGE output orography files first, then generates binary mask
+    of land points within the orography band specified.
+    """
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<GenerateOrographyBandAncils>')
+        return result
+
+    @staticmethod
+    def gen_orography_masks(
+            standard_orography, standard_landmask, key, thresholds):
+        """
+        Function to generate topographical band masks.
+
+        Parameters
+        -----------
+        standard_orography : cube
+            The standard orography, read from StaGE.
+        standard_landmask : cube
+            The landmask generated by gen_landmask.
+        key : string
+            Key from THRESHOLD_DICT which descibes type of topography band.
+        thresholds: float or list
+            Upper and/or lower thresholds of the current topographical band.
+
+        Returns
+        -------
+        mask_cube : cube
+            Cube containing topographical band mask.
+
+        Raises
+        ------
+        KeyError: if the key does not match any in THRESHOLD_DICT.
+        """
+        coords = standard_orography.coords()
+        if key == 'max_land_threshold':  # mask everything above max bound
+            orog_band = np.ma.masked_greater(
+                standard_orography.data, thresholds[0]).mask.astype(int)
+            mask_data = np.ma.masked_where(
+                np.logical_not(standard_landmask.data), orog_band)
+            sea_fillvalue = np.ma.default_fill_value(mask_data.data)
+            mask_data.data[mask_data.mask] = sea_fillvalue
+            mask_cube = _make_mask_cube(mask_data, key, coords,
+                                        lower_threshold=thresholds)
+        elif key == 'land':  # regular topographical bands above land
+            old_threshold, threshold = thresholds
+            orog_band = np.ma.masked_inside(
```

Comment:
I think that what's in `max_land_threshold` and `land` are quite similar. Would it be possible to factor any of it out?